2001-08-30 12:07  Henrik Nordstrom <hno@squid-cache.org>

        BUGFIX: potential coredump condition on snmpwalk of empty tables.
	If the indexing function did not return a new OID, make sure the
	parsing function is forgotten as well.

Index: squid/src/snmp_core.c
diff -u squid/src/snmp_core.c:1.45.2.2 squid/src/snmp_core.c:1.45.2.3
--- squid/src/snmp_core.c:1.45.2.2	Sun May 27 17:52:21 2001
+++ squid/src/snmp_core.c	Thu Aug 30 04:07:28 2001
@@ -731,7 +731,10 @@
 	*NextLen = mibTreeEntry->len;
 	*Next = (*mibTreeEntry->instancefunction) (mibTreeEntry->name, NextLen, mibTreeEntry, &Fn);
     }
-    return (Fn);
+    if (*Next)
+	return (Fn);
+    else
+	return NULL;
 }
 
 static oid *
